app/models/edition/identifiable.rb

```diff
@@ -16,10 +16,15 @@ def linkable?
   end
 
   def ensure_presence_of_document
-    self.document ||= Document.new(
-      sluggable_string: string_for_slug,
-      content_id: SecureRandom.uuid,
-    )
+    if document.blank?
+      self.document = Document.new(
+        sluggable_string: string_for_slug,
+        content_id: SecureRandom.uuid,
+      )
+    else
```

Comment:
maybe this should be 

```
elsif document.new_record?
 document.sluggable_string = string_for_slug
 document.content_id = SecureRandom.uuid 
end
```

wdyt?

Reply:
Yeah, I reckon we could make this more intention revealing and remove some repetition.

What about something like this?

```ruby
def ensure_presence_of_document
  build_document if document.nil?
  
  if document.new_record?
    document.sluggable_string = string_for_slug
    document.content_id = SecureRandom.uuid
  end
end
```